### PR TITLE
[WIP] Change Getting Started to use RSpec

### PR DESCRIPTION
**Pull request by cllns**
For now, using `hanami new bookshelf --test=rspec`, but we anticipate to change RSpec to be the default testing framework for Hanami.

MiniTest is simpler, but it's less familiar to most people.
Additionally, when this was created, Hanami itself used MiniTest.
We've since switched over to RSpec, since we needed some of its more powerful features (and it's what most of use most of the time).

**NOTE**: This is _not_ ready to merge. I need to update the code examples from running the test suite, since MiniTest's output is different than RSpec's. I also need to go through the guide again, to make sure I didn't miss anything else. Just thought I'd open this to push myself to finish this sooner, and maybe get another set of eyes on the changes :) 🌸

### CODE REVIEW

File: source/guides/1.1/getting-started.md [hunk 1]
```diff
@@ -230,35 +260,37 @@ module Web::Views::Home
...
-Let's use our new knowledge about the major Hanami components to add a new action.
-The purpose of our Bookshelf project is to manage books.
+Let's use our new knowledge about Hanami __routes__, __actions__, __views__, and __templates__.
+
+The purpose of our sample Bookshelf project is to manage books.
 
 We'll store books in our database and let the user manage them with our project.
-A first step would be to show a listing of all the books in our system.
+
+Our first step will be list out all the books we know about.
```
Comment (oana-sipos) [inline source/guides/1.1/getting-started.md]: Isn't it more correct to say: _Our first step will be *to* list out..._ ?

File: source/guides/1.1/getting-started.md [hunk 2]
```diff
@@ -271,36 +303,63 @@ describe 'List books' do
...
+
+- Creating an action at `apps/web/controllers/books/index.rb` (and spec for it),
+- Creating a view at `apps/web/views/books/index.rb` (and a spec for it),
+- Creating a template at `apps/web/templates/books/index.html.erb`.
+
+_(If you're confused by 'action' vs. 'controller', here's a tip:
+Hanami only has `action` classes,
+so a controller is just a module to group several related actions together.)_
+
+They files are all pretty much empty.
```
Comment (oana-sipos) [inline source/guides/1.1/getting-started.md]: Did you maybe mean **these files**?

**Comment by oana-sipos** [comment]
All good as far as I can see at first sight. A minor comment (and maybe just a preference from my side) is not to break a sentence on separate lines in `.md`, both editors and renderers know how to print it as part of the next line. Please ping me if it's unclear ;) Otherwise, nice job @cllns!

**Comment by cllns** [comment]
@oana-sipos Thanks for the review! The breaks mid-sentence are actually intentional, see [this post about 'Semantic Linefeeds'](http://rhodesmill.org/brandon/2012/one-sentence-per-line/). It helps make git diffs nicer, IMO. It's definitely a preference thing though.

**Comment by oana-sipos** [comment]
@cllns thank you for the perspective ;) nice read!

File: source/guides/1.1/getting-started.md [hunk 3]
```diff
@@ -347,42 +412,51 @@ Open up the file `apps/web/templates/application.html.erb` and edit it to look l
...
-* a **repository** mediates between entities and the persistence layer.
+* an **entity** is a domain object (a `Book`) that is uniquely identified by its identity,
+* a **repository** is what we use to persist, retrieve, and delete data for an entity, in the persistence layer.
 
 Entities are totally unaware of the database.
 This makes them **lightweight** and **easy to test**.
 
-For this reason we need a repository to persist the data that a `Book` depends on.
+Since entities are completely decoupled from the database,
+we use repositories to persist the data behind a `Book`.
```
Comment (GustavoCaso) [inline source/guides/1.1/getting-started.md]: Maybe we could add an external link to explain what is the repository pattern.

I have found these one https://martinfowler.com/eaaCatalog/repository.html

**Comment by jodosha** [comment]
@cllns Thanks for this PR. The contents has been ported to https://github.com/hanami/guides/pull/3

**Pull request closed without merging**